src/pytesmo/validation_framework/data_averaging.py

```diff
@@ -0,0 +1,423 @@
+
+        Parameters
+        ----------
+        df: pd.DataFrame
+            Dataframe of values to average using method
+        method: str
+            averaging method
+        kwargs: keyword arguments
+            Arguments for some upscaling functions
+
+        Returns
+        -------
+        upscaled: pandas.DataFrame
+            dataframe with "upscaled" column
+        """
+        up_function = {
+            "average": self.faverage,
+        }
+        f = up_function[method]
+        # check that column names are all the same and return first
+        g = groupby(df.columns)
+        assert (next(g, True) and not next(g, False))
+        var_name = df.columns[0]
+
+        upscaled = pd.DataFrame(
+            data=f(df, **kwargs),
+            columns=[var_name]
+        )
+
+        return upscaled
+
+    @staticmethod
+    def faverage(df) -> pd.Series:
+        """Simple average of each column in the dataframe"""
+        # workaround to avoid changes in place
+        out = df.mean(axis=1)
+
+        return out
+
+    def wrapper(
+            self,
+            gpi,
+            other_name,
+            tmatching="rescale",
+            up_method="average",
+            tstability=False,
+            **kwargs
+    ) -> pd.DataFrame:
+        """
+        Find the upscale estimate with given method, for a certain reference gpi
+
+        Parameters
+        ----------
+        gpi: int
+            gpi value of the reference point
+        other_name: str
+            name of the non-reference dataset to be upscaled
+        tmatching: str, default is "rescale"
+            method to use for temporal matching. 'rescale' uses the pytesmo.temporal_matching methods, 'common' takes
+            only measurements at common times between all the pms
+        up_method: str
+            method to use for upscaling:
+                * 'average' takes the simple mean of all timeseries
+        tstability: bool, default is False
+            if True, the values are filtered using the time stability concept
+        kwargs: keyword arguments
+            argumjents for the temporal window or time stability thresholds
+
+        Returns
+        -------
+        upscaled: pd.DataFrame
+            upscaled time series
+        """
+        other_lut = self.lut[other_name]
+        if gpi in other_lut.keys():
+            other_points = other_lut[gpi]
+        else:
+            return None
+        # check that there are points for specific reference gpi
+        if not other_points:
+            return None
+
+        tss = self.get_timeseries(points=other_points, other_name=other_name)
+        # check that timeseries were extracted from the points
+        if not tss:
+            return None
+        tss = [df for df in tss if not df is None]
+
+        # handle situation with single timeseries or all None
+        if len(tss) < 1:
+            if not tss:
+                return None
+            return tss[0]
```

Comment:
This should be:
```
if not tss:
    return None
```
The second return statement does not make sens, since `len(tss) < 1`, so `tss[0]` does not exist.